Stabilize data migration tests

mmaslankaprv opened the pull request:
This PR contains the following fixes: 
- Passing timeout to Admin client in tests to make retries effective
- Fail test when failure injector fails to stop Redpanda
- Handle `resource_migrated_error` correctly
- Reduce timeouts in migration routers to make them more reasonable and be able to shutdown timely

Fixes: CORE-14484

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.3.x
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->
- none

mmaslankaprv committed:
tests: pass timeout to admin in migration tests

Migration tests use failure injector which may suspend Redpanda nodes.
In order to make sure the test can retry the Admin API request we must
use shorter timeout value.

Setting the default Admin API timeout to value smaller than default 30
seconds to make retries effective.

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv commented:
/ci-repeat 1
dt-repeat=20
tests/rptest/tests/data_migrations_api_test.py

mmaslankaprv commented:
/ci-repeat 1
dt-repeat=20
tests/rptest/tests/data_migrations_api_test.py

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--3e2b4915-b6c4-4e65-8007-4586fcb48bd8-->
<!--TEST RESULTS-->
<!--ba0a415c-4b6e-4e83-bfa6-403628482974-->
<!--TEST RESULTS-->
<!--33556424-eb8c-4aa0-b919-fbfbd3d980b8-->
<!--TEST RESULTS-->
<!--cb4e94b7-de1a-4e2d-b6d3-2b1f9cec2bd0-->
<!--TEST RESULTS-->
<!--a76ea6c3-75d4-49ad-86a4-e87662b9a047-->
<!--TEST RESULTS-->
<!--96f5661a-9dec-4119-a230-93ced0dfc6cb-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75639">build#75639</a></summary>


| test_class            | test_method                          | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                             |
|:----------------------|:-------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DataMigrationsApiTest | test_creating_and_listing_migrations | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75639#019a53b8-a444-4fb6-bcee-4eb43b1eedac | FLAKY         | 36/40    | upstream reliability is '98.98403483309144'. current run reliability is '81.81818181818183'. drift is 17.16585 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations |
| DataMigrationsApiTest | test_creating_and_listing_migrations | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75639#019a53bf-6237-442d-bdeb-adbfb302e3aa | FLAKY         | 29/40    | upstream reliability is '100.0'. current run reliability is '57.692307692307686'. drift is 42.30769 and the allowed drift is set to 50. The test should PASS            | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations |
| DataMigrationsApiTest | test_higher_level_migration_api      | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75639#019a53b8-a444-4fb6-bcee-4eb43b1eedac | FLAKY         | 39/40    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS               | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_higher_level_migration_api      |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75665">build#75665</a></summary>


| test_class            | test_method                                     | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                                        |
|:----------------------|:------------------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DataMigrationsApiTest | test_creating_and_listing_migrations            | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75665#019a54dc-9719-44f9-af7d-2c43558c1346 | FLAKY         | 33/40    | upstream reliability is '98.86765746638359'. current run reliability is '70.83333333333334'. drift is 28.03432 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations            |
| DataMigrationsApiTest | test_creating_and_listing_migrations            | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75665#019a54dc-971a-4aae-8194-4319f6fae67c | FLAKY         | 31/40    | upstream reliability is '99.07735982966643'. current run reliability is '64.0'. drift is 35.07736 and the allowed drift is set to 50. The test should PASS              | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations            |
| DataMigrationsApiTest | test_creating_and_listing_migrations            | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75665#019a54dc-971b-452d-8507-be20931ced3d | FLAKY         | 33/40    | upstream reliability is '99.92846924177397'. current run reliability is '69.56521739130434'. drift is 30.36325 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations            |
| DataMigrationsApiTest | test_creating_and_listing_migrations            | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75665#019a54dc-971b-48d1-b8b5-6aafdf213332 | FLAKY         | 38/40    | upstream reliability is '96.07708189951823'. current run reliability is '90.47619047619048'. drift is 5.60089 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations            |
| DataMigrationsApiTest | test_creating_and_listing_migrations            | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75665#019a54dc-9721-4811-8d4b-288bf3c6d8ca | FLAKY         | 34/40    | upstream reliability is '98.86845827439888'. current run reliability is '75.0'. drift is 23.86846 and the allowed drift is set to 50. The test should PASS              | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations            |
| DataMigrationsApiTest | test_creating_and_listing_migrations            | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75665#019a54dc-9722-41c6-b263-16d725c6ae82 | FLAKY         | 31/40    | upstream reliability is '99.92846924177397'. current run reliability is '60.86956521739131'. drift is 39.0589 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations            |
| DataMigrationsApiTest | test_creating_and_listing_migrations            | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75665#019a54dc-9723-4c71-a3c0-4be028643271 | FLAKY         | 34/40    | upstream reliability is '98.86765746638359'. current run reliability is '75.0'. drift is 23.86766 and the allowed drift is set to 50. The test should PASS              | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations            |
| DataMigrationsApiTest | test_creating_and_listing_migrations            | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75665#019a54dc-9724-4807-a840-a07297c4f68e | FLAKY         | 37/40    | upstream reliability is '99.07801418439716'. current run reliability is '86.95652173913044'. drift is 12.12149 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations            |
| DataMigrationsApiTest | test_creating_and_listing_migrations            | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75665#019a54dc-9724-4ac5-a510-ff76ee77278c | FLAKY         | 33/40    | upstream reliability is '98.86925795053003'. current run reliability is '73.07692307692307'. drift is 25.79233 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations            |
| DataMigrationsApiTest | test_creating_and_listing_migrations_wo_license | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75665#019a54dc-971a-4aae-8194-4319f6fae67c | FLAKY         | 39/40    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS               | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations_wo_license |
| DataMigrationsApiTest | test_higher_level_migration_api                 | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75665#019a54dc-971a-44c7-9e41-64c54aece321 | FLAKY         | 39/40    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS               | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_higher_level_migration_api                 |
| DataMigrationsApiTest | test_higher_level_migration_api                 | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75665#019a54dc-971b-48d1-b8b5-6aafdf213332 | FLAKY         | 38/40    | upstream reliability is '99.3975903614458'. current run reliability is '90.9090909090909'. drift is 8.4885 and the allowed drift is set to 50. The test should PASS     | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_higher_level_migration_api                 |
| DataMigrationsApiTest | test_higher_level_migration_api                 | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75665#019a54dc-9724-4ac5-a510-ff76ee77278c | FLAKY         | 39/40    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS               | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_higher_level_migration_api                 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75735">build#75735</a></summary>


| test_class            | test_method                          | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                             |
|:----------------------|:-------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DataMigrationsApiTest | test_creating_and_listing_migrations | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75735#019a58c7-7894-4ce7-b3cc-90cf3ca34e55 | FLAKY         | 35/40    | upstream reliability is '98.84432358939496'. current run reliability is '77.27272727272727'. drift is 21.5716 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations |
| DataMigrationsApiTest | test_creating_and_listing_migrations | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75735#019a58cd-64f9-48c5-8a08-a631d8f31204 | FLAKY         | 36/40    | upstream reliability is '100.0'. current run reliability is '82.6086956521739'. drift is 17.3913 and the allowed drift is set to 50. The test should PASS              | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75762">build#75762</a></summary>


| test_class            | test_method                     | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason   | test_history                                                                                                                                                        |
|:----------------------|:--------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DataMigrationsApiTest | test_conflicting_migrations     | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75762#019a59b7-b6b7-44af-88a6-e30876f16a85 | FAIL          | 0/40     |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_conflicting_migrations     |
| DataMigrationsApiTest | test_conflicting_migrations     | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75762#019a59b8-3788-4723-9f81-fd379624e97b | FAIL          | 0/40     |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_conflicting_migrations     |
| DataMigrationsApiTest | test_higher_level_migration_api | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75762#019a59b7-b6b7-44af-88a6-e30876f16a85 | FLAKY         | 39/40    |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_higher_level_migration_api |
| DataMigrationsApiTest | test_mount_inexistent           | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75762#019a59b7-b6b7-44af-88a6-e30876f16a85 | FLAKY         | 39/40    |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_mount_inexistent           |
| DataMigrationsApiTest | test_mount_inexistent           | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75762#019a59b8-3788-4723-9f81-fd379624e97b | FLAKY         | 39/40    |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_mount_inexistent           |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75816">build#75816</a></summary>


| test_class                    | test_method                     | test_arguments                                                                    | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                       |
|:------------------------------|:--------------------------------|:----------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ReplicatedMetastoreTest       | TestBasicRemoveTopics           |                                                                                   | unit        | https://buildkite.com/redpanda/redpanda/builds/75816#019a5d5d-3fbc-49b2-9b57-2de727698ee3 | FAIL          | 0/1      |                                                                                                                                                                        |                                                                                                                                                                    |
| ShadowLinkingReplicationTests | test_replication_basic          | {"shuffle_leadership": true, "source_cluster_spec": {"cluster_type": "redpanda"}} | integration | https://buildkite.com/redpanda/redpanda/builds/75816#019a5d7e-097e-47b9-b958-f469e6ed144e | FLAKY         | 16/21    | upstream reliability is '94.9738219895288'. current run reliability is '76.19047619047619'. drift is 18.78335 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_basic |
| DataMigrationsApiTest         | test_conflicting_migrations     | null                                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/75816#019a5d7e-0980-4e7c-b86f-e35c011db09e | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                 | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_conflicting_migrations    |
| ScalingUpTest                 | test_moves_with_local_retention | {"use_topic_property": false}                                                     | integration | https://buildkite.com/redpanda/redpanda/builds/75816#019a5d7e-097b-4c79-a36f-40b66bc53e8a | FLAKY         | 20/21    | upstream reliability is '97.63313609467455'. current run reliability is '95.23809523809523'. drift is 2.39504 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ScalingUpTest&test_method=test_moves_with_local_retention        |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75819">build#75819</a></summary>


| test_class                    | test_method                          | test_arguments                                                                                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                             |
|:------------------------------|:-------------------------------------|:---------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingReplicationTests | test_topic_delete                    | {"source_cluster_spec": {"cluster_type": "kafka", "kafka_quorum": "COMBINED_KRAFT", "kafka_version": "3.8.0"}} | integration | https://buildkite.com/redpanda/redpanda/builds/75819#019a5e70-1b57-4425-8c31-bdc9141dbde8 | FLAKY         | 20/21    | upstream reliability is '98.19967266775778'. current run reliability is '95.23809523809523'. drift is 2.96158 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_topic_delete            |
| DataMigrationsApiTest         | test_creating_and_listing_migrations | null                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/75819#019a5e78-0e59-4a18-84b7-f15915470c74 | FLAKY         | 20/21    | upstream reliability is '99.92565055762081'. current run reliability is '95.23809523809523'. drift is 4.68756 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations |
</details>
<!--/body-->

mmaslankaprv commented:
/ci-repeat 5
skip-redpanda-build
dt-repeat=20
tests/rptest/tests/data_migrations_api_test.py

mmaslankaprv commented:
/ci-repeat 1
dt-repeat=20
tests/rptest/tests/data_migrations_api_test.py

mmaslankaprv committed:
e2e_finjector: propagate failure injection errors to test

Sometimes when node fails to stop timely when injecting terminate
failure type it may happen that the cluster is in weird state as the
node is supposed to be stopped while it is not.

Changed the code to propagate the failure injector error to the test and
finally fail the test itself.

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv commented:
/ci-repeat 1
dt-repeat=20
tests/rptest/tests/data_migrations_api_test.py

mmaslankaprv committed:
admin: map resource is being migrated to bad request

When external user request already migrated resource migration f.e.
topic that is already migrated it is a bad request not the internal
server error.

Added mapping of `resource_is_being_migrated` error code to `400
BadRequest` error code.

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv committed:
dm/router: use more reasonable timeout values in migration router

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv commented:
/ci-repeat 1
dt-repeat=20
tests/rptest/tests/data_migrations_api_test.py

mmaslankaprv commented:
/ci-repeat 1
dt-repeat=20
tests/rptest/tests/data_migrations_api_test.py

bashtanov commented on tests/rptest/utils/data_migrations.py:
```diff
@@ -102,9 +106,9 @@ def get_migrations_map(self, node=None, redpanda: RedpandaService | None = None)
-        redpanda.logger.debug("received self.admin.list_data_migrations result")
+        redpanda.logger.debug("calling self.admin().list_data_migrations")
+        migrations = self.get_admin(redpanda).list_data_migrations(node).json()
+        redpanda.logger.debug("received self.admin().list_data_migrations result")
```
nit: wrong function name in the log messages

bashtanov commented on tests/rptest/tests/e2e_finjector.py:
```diff
@@ -46,6 +47,7 @@ def __init__(self, redpanda, scale, **kwargs):
         self.custom_failures = []
         self.max_concurrent_failures = sys.maxsize
         self.configure_finjector(**kwargs)
+        self.error_queue = queue.Queue(1)
```
A queue sounds like over-engineering here,just a variable holding None or error would be sufficient. Since only the loop thread is going to write into it, it is thread-safe thanks to GIL.

bashtanov reviewed (dismissed):
LGTM in generatl, a couple of nits

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75816-->
<!--716d5760-ca51-4372-a163-f53120ed07d2-->
<!--title-->
## Retry command for [Build#75816](https://buildkite.com/redpanda/redpanda/builds/75816)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_conflicting_migrations
```
<!--/body-->
